### DISCUSSION TITLE
Supress Psalm Error in echo shorthand

### DISCUSSION THREAD

**Discussion by mathe42** [vimeo/psalm#10987] [Q&A]
Lets assume we have a function "example" and we have a rule "noExample" that forbiddes the use of that function.

```php
<?= example(1+1)?>
```

and in this case I want to supress the rule. How can I do this?

Thanks in advance for all help.

---
EDIT:

As my text is not quite clear here a simple example:

https://psalm.dev/r/ce301fb03e

I would expect that the example should work but it doesn't

**Comment by ghostwriter** [comment]
You can use the `<issueHandlers>` tag in the config file

[Config suppression](https://psalm.dev/docs/running_psalm/dealing_with_code_issues/#config-suppression)

You can also use `@psalm-suppress IssueName` docblock to suppress an issues

[Docblock suppression](https://psalm.dev/docs/running_psalm/dealing_with_code_issues/#docblock-suppression)

**Reply by mathe42** [reply]
I know that. That is not helpfull... See this example  https://psalm.dev/r/ce301fb03e - what do I have to do to supress that specific error with a docblock?

**Reply by jorgsowa** [reply]
You have a typo in your example. However it doesn't work anyways: https://psalm.dev/r/910d5b0fc1

**Reply by ghostwriter** [reply]
Inline-suppression is not working, however you should still be able to use the `<issueHandlers>` tag in the config file.

e.g.

```diff
<psalm>
    <projectFiles>
        <directory name="src"/>
        <directory name="tests"/>
    </projectFiles>
    <issueHandlers>
+        <UndefinedGlobalVariable>
+            <errorLevel type="suppress">
+                <!-- all UndefinedGlobalVariable issues in this directory are suppressed -->
+                <directory name="path/to/views/directory"/>
+                <!-- all UndefinedGlobalVariable issues in this file are suppressed -->
+                <file name="path/to/views/file.view.php"/>
+            </errorLevel>
+        </UndefinedGlobalVariable>
        <PropertyNotSetInConstructor>
            <errorLevel type="suppress">
                <directory name="tests"/>
            </errorLevel>
        </PropertyNotSetInConstructor>
    </issueHandlers>
</psalm>
```
